motion_tracker/utils/image_generator.py

```diff
@@ -4,212 +4,195 @@
-    empty_img_accumulator = np.zeros([0, output_width, output_height, 3]).astype('uint8')
-    empty_box_accumulator = np.zeros([0, 4]).astype('uint16')
-    empty_X = {'start_img': empty_img_accumulator,
-               'start_box': empty_box_accumulator,
-               'end_img': empty_img_accumulator}
-    empty_y = {'end_box': empty_box_accumulator}
-    return empty_X, empty_y
-
-def rough_normalization(img):
-    return (img-100) / 60
-
-def tf_th_dim_switch(img_array):
-    """Switches between 'tf' and 'th' dim ordering. Assumes img num is in axis 0
-
-    tf ordering has channel in axis 3, and th ordering has channel in axis 1
-    """
-    return img_array.swapaxes(1,3)
-
-
-def master_generator(crops_per_image = 10, batch_size=50,
-                     output_width = 256, output_height = 256,
-                     desired_dim_ordering='th'):
-    """Generator yielding dictionary of image and bounding boxes.
-
-    Args:
-    ----
-        crops_per_image: int
-            Number of random crops of the current frame to take.
-            Corresponds to both k3 and k4 in the paper, with a value of 10.
-        batch_size: int
-            Number of items in list to be returned.
-        output_width: int
-            Width in pixels of output images for start and ending image
-        output_height: int
-            Height in pixels of output images for start and ending image
 
-    """
+class ImagePairAndBoxGen(object):
+    def __init__(self, crops_per_image = 10, batch_size=50,
+                 output_width = 256, output_height = 256,
+                 desired_dim_ordering='th'):
+        """Generator yielding dictionary of image and bounding boxes.
+        This is an abstract class and will not be called directly.
+
+        Args:
+        ----
+            crops_per_image: int
+                Number of random crops of the current frame to take.
+                Corresponds to both k3 and k4 in the paper, with a value of 10.
+            batch_size: int
+                Number of items in list to be returned.
+            output_width: int
+                Width in pixels of output images for start and ending image
+            output_height: int
+                Height in pixels of output images for start and ending image
+            desired_dim_ordering: Either 'th' or 'tf' to determine where channel index is
+        """
+        self.crops_per_image = crops_per_image
+        self.batch_size = batch_size
+        self.output_width = output_width
+        self.output_height = output_height
+        self.desired_dim_ordering = desired_dim_ordering
+
+
+    def are_accumulators_full(self):
+        return self.X['start_img'].shape[0] >= self.batch_size
+
+    def flow(self):
+        """The function yielding an iterator."""
+        while True:
+            self.reset_accumulators()
+            while not self.are_accumulators_full():
+                self.add_to_accumulators()
+            self.remove_accumulator_overflow()
```

Comment:
This is a question.. I thought we decided that we weren't going to care about this, and let the generator just keep going, since the keras model just takes whatever it is given? Is this in here just in case we decide we want it back?